internal_workflow_testsuite.go

```diff
@@ -827,13 +936,56 @@ func (env *testWorkflowEnvironmentImpl) RegisterSignalHandler(handler func(name
 }
 
 func (env *testWorkflowEnvironmentImpl) RequestCancelWorkflow(domainName, workflowID, runID string) error {
-	env.workflowCancelHandler()
+	if env.workflowInfo.WorkflowExecution.ID == workflowID {
+		// cancel current workflow
+		env.workflowCancelHandler()
+
+		// check if current workflow is a child workflow
+		parentEnv := env.parentEnv
+		if parentEnv != nil && parentEnv.onChildWorkflowCanceledListener != nil {
+			parentEnv.postCallback(func() {
+				parentEnv.onChildWorkflowCanceledListener(env.workflowInfo)
+			}, false)
+		}
+	} else if childHandle, ok := env.childWorkflows[workflowID]; ok {
+		// current workflow is a parent workflow, and we are canceling a child workflow
+		delete(env.childWorkflows, workflowID)
```

Comment:
no, it should be in the else case, as we are deleting child from parent, so we do it when current workflow is parent workflow.